src/data-workspace/category-combo-table/category-combo-table.js

```diff
@@ -8,55 +8,63 @@ import {
 } from '@dhis2/ui'
 import cx from 'classnames'
 import PropTypes from 'prop-types'
-import React from 'react'
-import { useMetadata, selectors } from '../metadata/index.js'
-import { cartesian } from '../shared/utils.js'
-import styles from './category-combo-table.module.css'
+import React, { useMemo } from 'react'
+import { useMetadata, selectors } from '../../metadata/index.js'
+import { cartesian } from '../../shared/utils.js'
 import {
     DataEntryCell,
     DataEntryField,
     useActiveCell,
-} from './data-entry-cell/index.js'
+} from '../data-entry-cell/index.js'
+import styles from './category-combo-table.module.css'
+import { TotalHeader, ColumnTotals, RowTotal } from './total-cells.js'
 
 export const CategoryComboTable = ({
     categoryCombo,
     dataElements,
     filterText,
     globalFilterText,
     maxColumnsInSection,
+    renderRowTotals,
+    renderColumnTotals,
 }) => {
     const { data } = useMetadata()
     const { deId: activeDeId, cocId: activeCocId } = useActiveCell()
 
-    if (!data) {
-        return null
-    }
-
     const categories = selectors.getCategoriesByCategoryComboId(
         data,
         categoryCombo.id
     )
 
-    // each element is a combination of category-options for a particular column
-    // this results in lists of category-options in the same order as headers are rendered
-    // the result is a client side computation of categoryOption-combinations
-    const optionsIdLists = categories.map((cat) => cat.categoryOptions)
-    const computedCategoryOptions = cartesian(optionsIdLists)
+    const sortedCOCs = useMemo(() => {
```

Comment:
I did some refactoring here, because at first I rendered the total-cells in this component - and thus had to subscribe to values in this component - but that means re-rendering the entire table on every-keystroke. I abandoned that, but kept some of the refactoring.

I think we could look into also refactoring the table into `CategoryComboTableHeader` -component and move some of the logic out of this component.